From 9cec9ed2a87b4cba7a1092267456974f86aaf546 Mon Sep 17 00:00:00 2001
From: Kai Uwe Broulik <kde@privat.broulik.de>
Date: Fri, 13 May 2016 14:14:53 +0200
Subject: [PATCH 3/4] [KSplashQML] Don't wait for KWin to start on Wayland

On Wayland KWin is the first thing to start, so KSplash just times out after 30s

Differential Revision: https://phabricator.kde.org/D1606
---
 ksplash/ksplashqml/splashapp.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/ksplash/ksplashqml/splashapp.cpp b/ksplash/ksplashqml/splashapp.cpp
index f6b69d8..b3ca365 100644
--- a/ksplash/ksplashqml/splashapp.cpp
+++ b/ksplash/ksplashqml/splashapp.cpp
@@ -66,6 +66,10 @@ SplashApp::SplashApp(int &argc, char ** argv)
 
     setStage(QStringLiteral("initial"));
 
+    if (QGuiApplication::platformName().startsWith(QLatin1String("wayland"), Qt::CaseInsensitive)) {
+        setStage(QStringLiteral("wm"));
+    }
+
     QPixmap cursor(32, 32);
     cursor.fill(Qt::transparent);
     setOverrideCursor(QCursor(cursor));
-- 
2.7.4

